tests/devices/i19/test_hutch_conditional_shutter.py

```diff
@@ -0,0 +1,160 @@
+    hcs = HutchConditionalShutter(
+        fake_prefix, hutch_two_from_which_shutter_is_controlled, shutter_name
+    )
+    assertIsInstance(hcs, HutchConditionalShutter)
+
+
+# Functional Test
+async def test_shutter_operates_correctly_for_hutch_one_when_in_use(
+    eh1_shutter: HutchConditionalShutter,
+):
+    set_mock_value(eh1_shutter.hutch_state, "EH1")
+    set_mock_value(eh1_shutter.shutter.interlock.status, 0)
+    set_mock_value(eh1_shutter.shutter.status, ShutterState.OPEN)
+
+    await eh1_shutter.set(ShutterDemand.CLOSE)
+
+    assert await eh1_shutter.shutter.status.get_value() == ShutterState.CLOSED
+    mock_shutter_control = get_mock_put(eh1_shutter.shutter.control)
+    mock_shutter_control.assert_has_calls([call(ShutterDemand.CLOSE, wait=True)])
+
+
+# Functional Test
+async def test_shutter_operates_correctly_for_hutch_two_when_in_use(
+    eh2_shutter: HutchConditionalShutter,
+):
+    set_mock_value(eh2_shutter.hutch_state, "EH2")
+    set_mock_value(eh2_shutter.shutter.interlock.status, 0)
+    set_mock_value(eh2_shutter.shutter.status, ShutterState.OPEN)
+
+    await eh2_shutter.set(ShutterDemand.CLOSE)
+
+    assert await eh2_shutter.shutter.status.get_value() == ShutterState.CLOSED
+    mock_shutter_control = get_mock_put(eh2_shutter.shutter.control)
+    mock_shutter_control.assert_has_calls([call(ShutterDemand.CLOSE, wait=True)])
+
+
+# Happy Path Tests Above
+# Unhappy Path Tests Below
+
+
+# Constructor Test
+async def test_shutter_not_created_for_invalid_hutch():
+    fake_prefix = "I19-0"
+    not_a_hutch = HutchState.INVALID
+    shutter_name = "Shutter_that_never_was"
+    with pytest.raises(HutchInvalidError):
+        HutchConditionalShutter(fake_prefix, not_a_hutch, shutter_name)
+
+
+# Functional Test
+async def test_shutter_raises_error_on_set_if_hutch_invalid(
+    eh2_shutter: HutchConditionalShutter,
+):
+    set_mock_value(eh2_shutter.hutch_state, "INVALID")
+
+    with pytest.raises(HutchInvalidError):
+        await eh2_shutter.set(ShutterDemand.OPEN)
+
+
+# Functional Test
+@patch("dodal.devices.i19.shutter.LOGGER")
+async def test_shutter_does_not_operate_for_hutch_not_in_use_and_logs(
+    patch_log,
+    eh2_shutter: HutchConditionalShutter,
+):
+    set_mock_value(eh2_shutter.hutch_state, "EH1")
+    await eh2_shutter.set(ShutterDemand.OPEN)
+    patch_log.warning.assert_called_once()
+
+
+# Functional Test
+async def test_shutter_raises_error_for_hutch_in_use_but_not_safe_to_operate(
+    eh2_shutter: HutchConditionalShutter,
+):
+    set_mock_value(eh2_shutter.hutch_state, "EH2")
+    set_mock_value(eh2_shutter.shutter.interlock.status, 1)
+    assert await eh2_shutter.shutter.interlock.shutter_safe_to_operate() is False
+
+    with pytest.raises(ShutterNotSafeToOperateError):
+        await eh2_shutter.set(ShutterDemand.OPEN)
+
+
+# Tests above
+# Auxiliary functions below
+
+
+def assertIsInstance(an_instance, expected_type):
+    assert isinstance(an_instance, expected_type)
```

Comment:
Should: This should be snake case as a python function
```suggestion
def assert_is_instance(an_instance, expected_type):
    assert isinstance(an_instance, expected_type)
```

Reply:
Could: I would also personally argue that `assert_is_instance(x, y)` is no more readable than `assert isinstance(x, y)` and so feel this function is redundant but I don't feel that strongly.

Reply:
Might well be "no more readable" but it is more ophiomorphic ( Greek ophis = snake )